From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 30 Nov 2010 20:01:15 -0500
Subject: [scsi] qla4xxx: don't process devices untill probe done
Message-id: <20101130200114.10450.6127.sendpatchset@localhost.localdomain>
Patchwork-id: 29716
O-Subject: [RHEL 5.6 PATCH 25/28] qla4xxx: Do not process devices untill driver
	probe is done.
Bugzilla: 656999

Bugzilla
--------

Bug 656999 (https://bugzilla.redhat.com/show_bug.cgi?id=656999)

Upstream Status
---------------

To be sent upstream.

Description
-----------

>From 99ceb9503325a72e96ab56bbfbec4d4092db1f1a Mon Sep 17 00:00:00 2001
From: Prasanna Mumbai <prasanna.mumbai@qlogic.com>
Date: Tue, 23 Nov 2010 11:59:50 -0800
Subject: [PATCH 26/29] qla4xxx: Do not process devices untill driver probe is done.

Signed-off-by: Prasanna Mumbai <prasanna.mumbai@qlogic.com>

diff --git a/drivers/scsi/qla4xxx/ql4_init.c b/drivers/scsi/qla4xxx/ql4_init.c
index e16455b..a8e91ef 100644
--- a/drivers/scsi/qla4xxx/ql4_init.c
+++ b/drivers/scsi/qla4xxx/ql4_init.c
@@ -1477,6 +1477,11 @@ static void qla4xxx_add_device_dynamically(struct scsi_qla_host *ha,
 		return;
 	}
 
+	/* If driver probe has not yet completed, do not process
+	 * devices that have come back online or have gone away */
+	if (!test_bit(AF_PROBE_DONE, &ha->flags))
+		return;
+
 	if (qla4xxx_add_sess(ddb_entry, new_tgt)) {
 		DEBUG2(printk(KERN_WARNING
 			      "scsi%ld: failed to add new device at index "
